packages/react/focus-scope/src/FocusScope.tsx

```diff
@@ -26,44 +35,265 @@ type FocusScopeProps = {
 
 function FocusScope(props: FocusScopeProps) {
   const { children, trapped = false } = props;
-  const containerRef = React.useRef<HTMLElement>(null);
-  const focusScopeRef = React.useRef<ReturnType<typeof createFocusScope>>();
+  const [container, setContainer] = React.useState<HTMLElement | null>(null);
   const onMountAutoFocus = useCallbackRef(props.onMountAutoFocus);
   const onUnmountAutoFocus = useCallbackRef(props.onUnmountAutoFocus);
+  const lastFocusedElementRef = React.useRef<HTMLElement | null>(null);
+
+  const wrapped = trapped;
+  const contained = trapped;
+
+  const scopeRef = React.useRef({
+    paused: false,
+    pause() {
+      scopeRef.current = { ...scopeRef.current, paused: true };
+    },
+    resume() {
+      scopeRef.current = { ...scopeRef.current, paused: false };
+    },
+  });
```

Comment:
Feel free to ignore but thought it might simplify things a bit if we did `.current` here and maybe use `this` to reduce some boilerplate:

```jsx
const focusScope = React.useRef({
	paused: false,
	pause() { this.paused = true },
	resume() { this.paused = false }
}).current;
```

Then everywhere else can just do 

- `focusScope.pause()`
- `focusScope.resume()`
- `if (focusScope.paused) {`
- `activeFocusStack.pause()` 

etc...

Reply:
I like that, will update 👍